### ISSUE TITLE
Dropdowns open on alternate display

### ISSUE THREAD

**Issue by bertday** [qgis/QGIS#59970]
### What is the bug or the crash?

When I click a dropdown selector in the QGIS UI, such as setting the display name for a layer, the options are displayed floating on a different screen (MacBook monitor vs. external display). I've tried creating a new QGIS profile and I'm still getting the same result. It's a little hard to capture since it's across two screens, but I will try to drop some screenshots below.

**Main Display**

![image](https://github.com/user-attachments/assets/1965f783-7087-49ea-9172-234d39fb0607)

**Secondary Display**

<img width="760" alt="image" src="https://github.com/user-attachments/assets/5a89999f-ed71-469d-86ba-e140fc2ab02e" />

----

But then randomly it will work correctly...

![image](https://github.com/user-attachments/assets/7d8f0086-d595-403a-9828-85d620dbe01c)

### Steps to reproduce the issue

1. Go to the properties for a layer
2. Select any dropdown, such as the Display > Display Name

### Versions

QGIS version | 3.38.3-Grenoble | QGIS code revision | 37f9e6efeec
-- | -- | -- | --
Qt version | 5.15.2
Python version | 3.9.5
GDAL/OGR version | 3.3.2
PROJ version | 8.1.1
EPSG Registry database version | v10.028 (2021-07-07)
GEOS version | 3.9.1-CAPI-1.14.2
SQLite version | 3.35.2
PDAL version | 2.3.0
PostgreSQL client version | unknown
SpatiaLite version | 5.0.1
QWT version | 6.1.6
QScintilla2 version | 2.11.5
OS version | macOS 14.6
  |   |   |  
Active Python plugins
zoom_level | 0.1
quick_map_services | 0.19.33
nominatim | 1.4.5
latlontools | 3.6.11
processing | 2.12.99
grassprovider | 2.12.99
db_manager | 0.1.20
MetaSearch | 0.3.6

</body></html>QGIS version
3.38.3-Grenoble
QGIS code revision
[37f9e6efeec](https://github.com/qgis/QGIS/commit/37f9e6efeec)
Qt version
5.15.2
Python version
3.9.5
GDAL/OGR version
3.3.2
PROJ version
8.1.1
EPSG Registry database version
v10.028 (2021-07-07)
GEOS version
3.9.1-CAPI-1.14.2
SQLite version
3.35.2
PDAL version
2.3.0
PostgreSQL client version
unknown
SpatiaLite version
5.0.1
QWT version
6.1.6
QScintilla2 version
2.11.5
OS version
macOS 14.6

Active Python plugins
zoom_level
0.1
quick_map_services
0.19.33
nominatim
1.4.5
latlontools
3.6.11
processing
2.12.99
grassprovider
2.12.99
db_manager
0.1.20
MetaSearch
0.3.6

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

**Comment by bertday** [comment]
Seems to be reported on StackExchange [here](https://gis.stackexchange.com/questions/430082/qgis-context-menu-always-opens-on-different-screen)

**Comment by nk9** [comment]
I'm experiencing this as well. It's pretty maddening, since it often appears that a click has been ignored. It's only when I look on the next screen over that I see that the window/menu has actually been opened, just hilariously misplaced.

**Comment by nk9** [comment]
I just noticed this same behavior on [DB Browser for SQLite](https://sqlitebrowser.org/). This is also a Qt application, so this would seem to be a framework bug. Perhaps we should look for a duplicate in the Qt bug tracker?

**Comment by nirvn** [comment]
This could well have been fixed in Qt6. 

@m-kuhn , are there vpckg macos packages people could test to confirm this issue has been resolved in Qt6?

**Comment by m-kuhn** [comment]
Yes, these can be downloaded here: https://github.com/qgis/QGIS/pull/60039#issuecomment-2567661209
